src/non_blocking/deflate/decode.rs

```diff
@@ -245,6 +245,12 @@ impl BlockDecoder {
             self.offset = new_len;
         }
     }
+
+    fn extend(&mut self, buf: &[u8]) {
+        self.buffer.extend(buf);
```

Comment:
I believe `extend_from_slice()` would be more efficient since it compiles to a `memcpy()`

https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice

Reply:
You're right. I will modify the code to use `extend_from_slice()`.

Reply:
Fixed: https://github.com/sile/libflate/commit/3a2ffb356e5fa63086ebb1cc1596f321b8c1dc71